drivers/led_strip/ws2812_spi.c

```diff
@@ -185,7 +191,16 @@ static int ws2812_spi_init(const struct device *dev)
 		}
 	}
 
-	return 0;
+	/* Setting the level of the SDO line to logic low to perform a reset before the driver is
+	 * available. This prevents weird boot behaviour of the leds caused by the logic level of
+	 * the SDO line during boot of the controller being interpreted as data signals.
+	 */
+	cfg->px_buf[0] = 0;
+
+	rc = spi_write_dt(&cfg->bus, &tx);
+	ws2812_reset_delay(cfg->reset_delay);
```

Comment:
Hi

As far as I understand, `spi-cpha` and `spi-cpol` do only affect the clock which is not used in this driver. This may indeed be a soc specific problem. I will create an issue regarding this pr.

Reply:
> As far as I understand, `spi-cpha` and `spi-cpol` do only affect the clock which is not used in this driver.

That's absolutely correct but please try it anyway. I've seen SPI mode affect the MOSI signal on some SoCs.

Reply:
I created https://github.com/zephyrproject-rtos/zephyr/issues/66372

Reply:
I tried all possible combinations of the `spi-cpha` and `spi-cpol` properties, but none of them seems to affect the mosi signal.

Reply:
Thanks